sdk/core/System.ClientModel/gen/TypeSymbolTypeRefCache.cs

```diff
@@ -62,6 +64,24 @@ private TypeRef FromTypeSymbol(ITypeSymbol symbol, TypeSymbolKindCache symbolToK
             }
         }
 
+        private static string? GetExperimentalDiagnosticId(ITypeSymbol symbol)
+        {
+            foreach (var attribute in symbol.GetAttributes())
+            {
+                if (attribute.AttributeClass?.Name == "ExperimentalAttribute" &&
```

Comment:
For efficiency use symbol matching similar to

https://github.com/Azure/azure-sdk-for-net/pull/51374/files#diff-c2ba6ff6c352287301ee394204c983ebce7e68e2f3e461b5d578022ad6bdb44aR99-R116